### DISCUSSION TITLE
Unable to figure out why 'LittleFS' was not declared in this scope

### DISCUSSION THREAD

**Discussion by the-programmer** [khoih-prog/ESPAsync_WiFiManager#96] [Q&A] [answered]
Hi all,

For a project I am trying to use the ESPAsync_WiFiManager but all the examples seem to be getting the same error.
The example I am using is [Async_ConfigOnDoubleReset](https://github.com/khoih-prog/ESPAsync_WiFiManager/blob/master/examples/Async_ConfigOnDoubleReset/Async_ConfigOnDoubleReset.ino)
This generates the following error. 
`src/main.cpp:87:30: error: 'LittleFS' was not declared in this scope`

It does not seem to be the same as issue #95 since I am using 2.0.3 as framework.

The [Build log.txt](https://github.com/khoih-prog/ESPAsync_WiFiManager/files/9082565/Build.log.txt) is attached.

Does anyone have any suggestions on how to fix this?

**Comment by khoih-prog** [comment, marked as answer]
Check [**Prerequisites**](https://github.com/khoih-prog/ESPAsync_WiFiManager#prerequisites)

> Notice: This [LittleFS_esp32 library](https://github.com/lorol/LITTLEFS) has been integrated to Arduino [ESP32 core v1.0.6+](https://github.com/espressif/arduino-esp32/tree/master/libraries/LITTLEFS) and you don't need to install it if using ESP32 core v1.0.6+

As you're using the new ESP32 cores v2.0.3 with built-in **LIttleFS**, you don't need [Lorol's **LITTLEFS** library](https://github.com/lorol/LITTLEFS) anymore. Especially you're using the recent PIO version supporting new ESP32 core v1.0.6+

You can try to comment out the following line in [**platformio.ini**](https://github.com/khoih-prog/ESPAsync_WiFiManager/blob/master/platformio/platformio.ini)

https://github.com/khoih-prog/ESPAsync_WiFiManager/blob/41f6479e85a9954b22afe9f4b816f96780e3d916/platformio/platformio.ini#L45

If not OK, try to compile using Arduino IDE to see if there is anything wrong with your installation. Then moving on to PIO.
Post the issue on **PIO Forum** for help if the problem persists.

Good Luck,

**Reply by the-programmer** [reply]
Just to let you know, In Arduino the library compiles without issues. However the following library's need to be installed manually. (And that is not super clear).
[ESPAsyncWebServer](https://github.com/me-no-dev/ESPAsyncWebServer)
[AsyncTCP](https://github.com/me-no-dev/AsyncTCP)

**Reply by the-programmer** [reply]
Also, a link to the question in the [PIO Forum](https://community.platformio.org/t/unable-to-remove-littlefs-library-from-espasync-wifimanager/28805).

**Reply by khoih-prog** [reply]
From [PIO Forum](https://community.platformio.org/t/unable-to-remove-littlefs-library-from-espasync-wifimanager/28805/3)

> |   |-- ESP_DoubleResetDetector @ 1.3.1
> |   |   |-- LittleFS_esp32 @ 1.0.6

As you're using new ESP32 core v1.0.6+, I think that you have to do something similar to [**fatal error: vfs_api.h: No such file or directory** #11](https://github.com/khoih-prog/ESP_WiFiManager_Lite/issues/11), such as remove all instances of `LittleFS_esp32` from `platformio.ini` and `library.json` of `ESP_DoubleResetDetector` library.

**Reply by the-programmer** [reply]
For reference. I just got all the patches correct to get `Async_ConfigOnDoubleReset.ino` working.
For `.pio\libdeps\esp32dev\ESP_DoubleResetDetector\library.json`
``` Diff
27,32d26
<     {
<       "owner": "lorol",
<       "name": "LittleFS_esp32",
<       "version": "^1.0.6",
<       "platforms": ["espressif32"]
<     }
```

For `.pio\libdeps\esp32dev\ESPAsync_WiFiManager\library.json`
``` Diff
49,54d48
<     },
<     {
<       "owner": "lorol",
<       "name": "LittleFS_esp32",
<       "version": ">=1.0.6",
<       "platforms": ["espressif32"]
```

For `.pio\libdeps\esp32dev\ESP Async WebServer\src\ESPAsyncWebServer.h`
``` Diff
27c27,28
< #include "FS.h"
---
> //#include "FS.h"
> #include <LittleFS.h>
```

Edit: The `.pio\libdeps\esp32dev\LittleFS_esp32` needs to be removed as well.

Hope this helps someone
